### PR TITLE
ISSUE-565 JMAP CalendarEvent/parse implementation for event title and description

### PR DESCRIPTION
**Pull request by vttranlina**
resolve https://github.com/linagora/tmail-backend/issues/565

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
the error maybe comes from LF/CRLF format of ics file 
I will check it

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala [hunk 1]
```diff
@@ -40,4 +47,452 @@ trait LinagoraCalendarEventParseMethodContract {
...
+           |        "accountId": "$ANDRE_ACCOUNT_ID",
+           |        "notFound": [ "$blobId" ]
+           |    },
+           |    "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def parseShouldFailWhenNumberOfBlobIdsTooLarge(): Unit = {
+    val blogIds: Array[String] = Range.inclusive(1, 999)
```
Comment (Arsnael) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:445]:
```suggestion
    val blobIds: Array[String] = Range.inclusive(1, 999)
```

**Comment by Arsnael** [comment]
```
[ERROR]   MemoryLinagoraCalendarEventParseMethodTest.parseShouldSuccess JSON documents are different:

Different keys found in node "methodResponses[0][1]", missing: "methodResponses[0][1].parsed", extra: "methodResponses[0][1].notParsable", expected: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","parsed":{"uploads-17c84a61-a7d6-4109-99a3-0d24db83e58e":{"description":"description 123","title":"Sprint planning #23"}}}> but was: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","notParsable":[uploads-17c84a61-a7d6-4109-99a3-0d24db83e58e]}>



[ERROR]   MemoryLinagoraCalendarEventParseMethodTest.parseShouldSuccessWhenMixSeveralCases JSON documents are different:

Different keys found in node "methodResponses[0][1]", missing: "methodResponses[0][1].parsed", expected: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","notFound":[253ff0ba-f490-47e2-aa5b-0599364e25c9],"notParsable":[uploads-2665f59c-d792-4f69-9ff3-570a17c03387],"parsed":{"uploads-7c1e3d73-dc83-4404-9b12-ccf2ac2a6066":{"description":"description 123","title":"Sprint planning #23"}}}> but was: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","notFound":[253ff0ba-f490-47e2-aa5b-0599364e25c9],"notParsable":[uploads-2665f59c-d792-4f69-9ff3-570a17c03387, uploads-7c1e3d73-dc83-4404-9b12-ccf2ac2a6066]}>

Array "methodResponses[0][1].notParsable" has different length, expected: <1> but was: <2>.

Array "methodResponses[0][1].notParsable" has different content. Extra values: ["uploads-7c1e3d73-dc83-4404-9b12-ccf2ac2a6066"], expected: <["uploads-2665f59c-d792-4f69-9ff3-570a17c03387"]> but was: <["uploads-2665f59c-d792-4f69-9ff3-570a17c03387","uploads-7c1e3d73-dc83-4404-9b12-ccf2ac2a6066"]>



[ERROR]   MemoryLinagoraCalendarEventParseMethodTest.parseShouldSupportSeveralBlobIds JSON documents are different:

Different keys found in node "methodResponses[0][1]", missing: "methodResponses[0][1].parsed", extra: "methodResponses[0][1].notParsable", expected: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","parsed":{"uploads-25954c54-55ac-4d34-9209-6c6bd973ca03":{"description":"description 456","title":"Sprint planning #24"},"uploads-bbbbc8ff-94a3-4c83-b00e-b7173b3c6c4d":{"description":"description 123","title":"Sprint planning #23"}}}> but was: <{"accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6","notParsable":[uploads-bbbbc8ff-94a3-4c83-b00e-b7173b3c6c4d, uploads-25954c54-55ac-4d34-9209-6c6bd973ca03]}>
```

**Review by Arsnael — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala [hunk 2]
```diff
@@ -40,4 +47,452 @@ trait LinagoraCalendarEventParseMethodContract {
 
     assertThat(response).contains("\"com:linagora:params:calendar:event\":{}")
   }
+
+  @Test
+  def parseShouldSuccess(): Unit = {
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:52]:
```suggestion
  def parseShouldSucceed(): Unit = {

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala [hunk 3]
```diff
@@ -40,4 +47,452 @@ trait LinagoraCalendarEventParseMethodContract {
...
+           |    "CalendarEvent/parse",
+           |    {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "notParsable": [ "$notParsableBlobId" ]
+           |    }, "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def parseShouldSuccessWhenMixSeveralCases(): Unit = {
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:235]:
```suggestion
  def parseShouldSucceedWhenMixSeveralCases(): Unit = {

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala [hunk 4]
```diff
@@ -40,4 +47,452 @@ trait LinagoraCalendarEventParseMethodContract {
...
+           |    "error",
+           |    {
+           |        "type": "accountNotFound"
+           |    },
+           |    "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def parseShouldNotParseBlobEntryWhenDoesNotPermission(server: GuiceJamesServer): Unit = {
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:399]:
```suggestion
  def parseShouldNotParseBlobEntryWhenDoesNotHavePermission(server: GuiceJamesServer): Unit = {

Comment (quantranhong1999) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:399]: Should we have a test for delegation?

Comment (chibenwa) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:399]: For shared mailbox also?

Comment (vttranlina) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala:399]: the ICS blob is not reference to the mailbox,
I don't find a way how to test it

Comment (chibenwa) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventParseMethodContract.scala]:
``` 
GIVEN bob shares his mailbox abc to Alice
AND email1 in ABC have an ICS attachment
THEN Alice should the able to EventCalendar/parse this attachment
```

**Review by Arsnael — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**